common/src/worldgen.rs

```diff
@@ -181,22 +181,22 @@ pub struct ChunkParams {
 }
 
 impl ChunkParams {
-    /// Extract data necessary to generate a chunk
-    ///
-    /// Returns `None` if an unpopulated node is needed.
-    pub fn new(dimension: u8, graph: &Graph, chunk: ChunkId) -> Option<Self> {
-        let state = graph[chunk.node].state.as_ref()?;
-        Some(Self {
+    /// Extract data necessary to generate a chunk, generating new graph nodes if necessary
+    pub fn new(dimension: u8, graph: &mut Graph, chunk: ChunkId) -> Self {
```

Comment:
Can we remove the `dimension` argument in favor of fetching it from the graph directly?